nncf/initialization.py

```diff
@@ -106,38 +122,102 @@ def write(cls, msg):
         stream_handler = logging.StreamHandler(TQDMStream)
         nncf_logger.addHandler(stream_handler)
 
-        initializers = OrderedDict()
-        hook_handles = []
-        for name, data in self.modules_to_init.items():
-            module, init_config = data
-            initializers[name] = RangeInitializerFactory.create(init_config, module, log_module_name=name)
-            hook_handles.append(module.register_forward_hook(initializers[name].forward_hook))
-
+        self._prepare_initialization()
         device = next(self.model.parameters()).device
-
         data_loader = wrap_dataloader_for_init(data_loader)
-        with torch.no_grad():
-            bar_format = '{l_bar}{bar} |{n_fmt}/{total_fmt} [{elapsed}<{remaining}]'
-            bar_desc = 'Algorithm initialization'
-            for i, loaded_item in tqdm(enumerate(data_loader), total=num_init_steps,
-                                       desc=bar_desc, bar_format=bar_format):
-                if num_init_steps is not None and i >= num_init_steps:
-                    break
-
-                args_kwargs_tuple = data_loader.get_inputs(loaded_item)
-                to_device_fn = partial(torch.Tensor.to, device=device)
-                args, kwargs = objwalk(args_kwargs_tuple, is_tensor, to_device_fn)
-                self.model(*args, **kwargs)
 
+        with torch.no_grad():
+            self._run_model_inference(data_loader, num_init_steps, device)
             nncf_logger.removeHandler(stream_handler)
-            for handle in hook_handles:
-                handle.remove()
-            for initializer in initializers.values():
-                initializer.apply_init()
-
+            self._apply_initializers()
 
         self.model.to(original_device)
+
+    def _prepare_initialization(self):
+        raise NotImplementedError
+
+    def _apply_initializers(self):
+        raise NotImplementedError
+
+
+class DataLoaderRangeInitializeRunner(DataLoaderBaseRunner):
+    def __init__(
+            self,
+            model,
+            modules_to_init_vs_init_configs: Dict[str, Tuple[torch.nn.Module, Dict]],
+            init_device: str,
+    ):
+        super().__init__(model, init_device)
+        self.modules_to_init = modules_to_init_vs_init_configs
+        self.progressbar_description = 'Quantizer range initialization'
```

Comment:
Quantization specifics shouldn't appear in a base class

Reply:
Renamed appropriately.